examples/1.0.0/bnpl-arazzo.yaml

```diff
@@ -0,0 +1,252 @@
+      - If the list of qualifying products is empty then end the workflow
+    operationId: findEligibleProducts
+    parameters:
+      - name: products
+        in: body
+        value: $inputs.products
+      - name: customer
+        in: body
+        value: $inputs.customer.uri
+    successCriteria:
+      - condition: $statusCode == 200
+    onSuccess:
+      - name: existingCustomerNotEligible
+        type: end
+        criteria:
+          - condition: $statusCode == 200
+          - condition: $response.body.existingCustomerNotEligible == false
+      - name: qualifyingProductsFound
+        type: goto
+        stepId: getCustomerTermsAndConditions
+        criteria:
+          - condition: $statusCode == 200
+          - context: $response.body
+            type: jsonpath
+            condition: $[?count(@.products) > 0]
+      - name: qualifyingProductsNotFound
+        type: end
+        criteria:
+          - condition: $statusCode == 200
+          - context: $response.body
+            type: jsonpath
+            condition: $[?count(@.products) == 0]
+    outputs:
+      eligibilityCheckRequired: $response.body.eligibilityCheckRequired
+      eligibleProducts: $response.body.products
+      totalLoanAmount: $response.body.totalAmount
+  - stepId: getCustomerTermsAndConditions
+    description: |
+      Get the terms and conditions for the BNPL loans. This is static data and therefore has no arguments.
+
+      The data will be displayed to the customer and they'll accept the terms out-of-band.
+
+      After this step the flow will need to do a customer eligibility check if required.
+    operationId: getTermsAndConditions
+    successCriteria:
+      - condition: $statusCode == 200
+    onSuccess:
+      - name: eligibilityCheckRequired
+        type: goto
+        stepId: createCustomer
+        criteria:
+        - condition: $steps.checkLoanCanBeProvided.outputs.eligibilityCheckRequired == true
+      - name: eligibilityCheckNotRequired
+        type: goto
+        stepId: initiateBnplTransaction
+        criteria:
+        - condition: $steps.checkLoanCanBeProvided.outputs.eligibilityCheckRequired == false
+    outputs:
+      termsAndConditions: $response.body
+  - stepId: createCustomer
+    description: |
+      Call the BNPL platform and verify the customer is eligible for the loan, which creates a customer resource. This step is skipped if the customer is already enrolled in the BNPL platform.
+
+      Accepting the terms and conditions is set to true as the assumption is they have been accepted when this step is invoked.
+
+      If the customer is eligible for the BNPL loan then a customer resource is created 
+    operationId: createCustomer
+    parameters:
+      - name: customerAttributes
+        in: body
+        value: $inputs.customer
+      - name: termsAndConditionsAccepted
+        in: body
+        value: true
+    successCriteria:
```

Comment:
When defining multiple success criteria, ALL must be satisfied for the step to be deemed successful. The way you have laid out the conditions here will result in the step never being successful as a statusCode cannot be both `200` and `201` at the same time.

Reply:
OK I misunderstood the Specification. Have accepted the suggestion